state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -1006,7 +932,90 @@ fn witness_broadcast(tx_out_id: [u8; 4]) {
 		MOCK_TX_METADATA,
 	));
 }
-enum TxType {
-	Normal,
-	Rotation,
+
+#[test]
+fn timed_out_broadcaster_are_reported() {
+	let mut expiry = 0u64;
+	new_test_ext()
+		.execute_with(|| {
+			let (broadcast_id, _) = start_mock_broadcast();
+			expiry = System::block_number()
+				.saturating_add(<Test as crate::Config<Instance1>>::BroadcastTimeout::get());
+			let nominee = AwaitingBroadcast::<Test, Instance1>::get(broadcast_id).unwrap().nominee;
+
+			assert!(FailedBroadcasters::<Test, Instance1>::get(broadcast_id).is_empty());
+			(broadcast_id, nominee)
+		})
+		.then_execute_at_block(expiry, |(broadcast_id, nominee)| {
+			// The nominated broadcaster is added to `FailedBroadcasters` to be reported later.
+			assert_eq!(
+				FailedBroadcasters::<Test, Instance1>::get(broadcast_id),
+				BTreeSet::from([nominee])
+			);
+		});
+}
+
+#[test]
+fn broadcast_can_be_aborted_due_to_time_out() {
+	let mut expiry = 0;
+	new_test_ext()
+		.execute_with(|| {
+			let (broadcast_id, _) = start_mock_broadcast();
+			expiry = System::block_number()
+				.saturating_add(<Test as crate::Config<Instance1>>::BroadcastTimeout::get());
+			ready_to_abort_broadcast(broadcast_id);
+
+			broadcast_id
+		})
+		.then_execute_at_block(expiry, |broadcast_id| broadcast_id)
+		.then_execute_with(|broadcast_id| {
+			// Broadcast should be aborted
+			System::assert_last_event(RuntimeEvent::Broadcaster(
+				crate::Event::<Test, Instance1>::BroadcastAborted { broadcast_id },
+			));
+			assert!(AbortedBroadcasts::<Test, Instance1>::get().contains(&broadcast_id));
+			assert_eq!(FailedBroadcasters::<Test, Instance1>::decode_len(broadcast_id), None);
+		});
+}
+
+#[test]
+fn aborted_broadcasts_can_still_succeed() {
+	let mut expiry = 0;
+	new_test_ext()
+		.execute_with(|| {
+			let (broadcast_id, transaction_out_id) = start_mock_broadcast();
+			expiry = System::block_number()
+				.saturating_add(<Test as crate::Config<Instance1>>::BroadcastTimeout::get());
+			ready_to_abort_broadcast(broadcast_id);
+
+			(broadcast_id, transaction_out_id)
+		})
+		.then_execute_at_block(expiry, |(broadcast_id, transaction_out_id)| {
+			(broadcast_id, transaction_out_id)
+		})
+		.then_execute_with(|(broadcast_id, transaction_out_id)| {
+			// Broadcast should be aborted
+			assert_eq!(FailedBroadcasters::<Test, Instance1>::decode_len(broadcast_id), None);
```

Comment:
We should add an assert to check that the AbortedBroadcasts has an item or check the event, just to confirm that it's aborted (and not just the FailedBroadcasters is empty - which would usually suggest it's not even been attempted yet and therefore not aborted